corehq/apps/reports/standard/deployments.py

```diff
@@ -234,6 +247,38 @@ def process_rows(self, users, fmt_for_export=False):
             ])
         return rows
 
+    def process_users(self, users, fmt_for_export=False):
+        rows = []
+        for user in users[self.pagination.start:self.pagination.end]:
+            rows.append([
+                user_display_string(user.user_dim.username,
+                                    user.user_dim.first_name,
+                                    user.user_dim.last_name),
+                _fmt_date(user.last_form_submission_date, fmt_for_export),
+                _fmt_date(user.last_sync_log_date, fmt_for_export),
+                self.get_app_name(user.app_dim.app_id) or '---',
+                user.last_form_app_build_version,
+                user.last_form_app_commcare_version
+            ])
+
+    def get_sql_sort(self):
+        res = []
+        #the NUMBER of cols sorting
+        sort_cols = int(self.request.GET.get('iSortingCols', 0))
+        if sort_cols > 0:
+            for x in range(sort_cols):
+                col_key = 'iSortCol_%d' % x
+                sort_dir = self.request.GET['sSortDir_%d' % x]
+                col_id = int(self.request.GET[col_key])
+                col = self.headers.header[col_id]
+                if col.sql_col is not None:
+                    res = col.sql_col
```

Comment:
Should this be `res.append(col.sql_col)`?

If yes, two lines below should be `res[-1] = '-{}'.format(res[-1])`

Reply:
Good catch. How would you feel if I instead had it only include the first sort column specified (and therefore leave it as a string, just break the loop)? I do not think there is a way in the report UI for a user to specific sorting on multiple columns and it feels like an unnecessary performance hit to the sql if we sort on multiple columns, but I am not sure it makes enough of a difference either way to matter.

Reply:
Sorting on a single column is fine if that meets requirements. Should probably change `res = []` to `res = ""` ([line 265](https://github.com/dimagi/commcare-hq/pull/19982/files/326fd18b3a7e1ce30b0d859895b4cb888ec4626b#diff-84a239b50f9046e07ef02b97cba3f1baR265)) and change `if len(res) == 0:` to `if not res:` ([line 278](https://github.com/dimagi/commcare-hq/pull/19982/files/326fd18b3a7e1ce30b0d859895b4cb888ec4626b#diff-84a239b50f9046e07ef02b97cba3f1baR278)).